runtime/lua/vim/_editor.lua

```diff
@@ -871,8 +871,8 @@ function vim._init_default_mappings()
   map('n', '<C-L>', '<Cmd>nohlsearch<Bar>diffupdate<Bar>normal! <C-L><CR>')
   map('i', '<C-U>', '<C-G>u<C-U>')
   map('i', '<C-W>', '<C-G>u<C-W>')
-  map('x', '*', 'y/\\V<C-R>"<CR>')
-  map('x', '#', 'y?\\V<C-R>"<CR>')
+  map('x', '*', [[y/\V<C-R>=escape(@", '/\')<CR>]])
+  map('x', '#', [[y?\V<C-R>=escape(@", '?\')<CR>]])
```

Comment:
I just noticed that these mappings are peforming a yank, wtf. 🤦 

Instead of another half-measure, may I request that we implement a full solution, as I linked to before: https://github.com/justinmk/config/blob/6e591bad58db8200e8eeebe20982ce3d1fc09c91/.config/nvim/init.vim#L794-L816

The current mappings also don't set `@/`, but they should.

Reply:
Would it be better to yank to `/`?

Reply:
that would be an improvement!

Reply:
I was guessing out loud earlier, and after some trying I don't think it's possible to `"/y` like I thought.

I'll look at the vim script you linked and make something like it in lua.

Reply:
Shouldn't `*` in visual mode also add work boundaries, `\<` and `\>`? vim-asterisk does this

Reply:
> Shouldn't `*` in visual mode also add work boundaries, `\<` and `\>`? vim-asterisk does this

The characters after `\<` and before `\>` must be word characters.

Reply:
> I just noticed that these mappings are peforming a yank, wtf. 🤦
> 
> Instead of another half-measure, may I request that we implement a full solution, as I linked to before: https://github.com/justinmk/config/blob/6e591bad58db8200e8eeebe20982ce3d1fc09c91/.config/nvim/init.vim#L794-L816

Correct me if I'm wrong, but this doesn't work when a multibyte character is the last one in a selection.  It only gets the first byte of the last character.

> 
> The current mappings also don't set `@/`, but they should.

The `/` register gets modified with the search command (right after the yank).

Reply:
> Correct me if I'm wrong, but this doesn't work when a multibyte character is the last one in a selection. It only gets the first byte of the last character.

idk, improvements welcome.

Reply:
Could something be built from
```vim
function! VisualSelection() abort
  try
    let unnamed_reg = @"
    let yank_reg = @0
    noautocmd silent! normal! gvy
    return @"
  finally
    let @" = unnamed_reg
    let @0 = yank_reg
  endtry
endfunction

xnoremap * <ESC>/\V<C-R>=substitute(escape(VisualSelection(), '/\'),'\n','\\n','g')<CR><CR>
xnoremap # <ESC>?\V<C-R>=substitute(escape(VisualSelection(), '?\'),'\n','\\n','g')<CR><CR>
```
(inspired from [#issuecomment-904514301](https://github.com/neovim/neovim/issues/15471#issuecomment-904514301) and [#issuecomment-1397813214](https://github.com/neovim/neovim/issues/21676#issuecomment-1397813214)) ?
It seems to work.

Reply:
That performs a yank, which is what the maintainers want to avoid.

Reply:
Yes, but here the different affected registers are saved and restored, which is not the case in version 0.9.0 of Neovim. I believe, unless I'm mistaken, that this is what can primarily bother the maintainers.